envctl/env_scripts/java/appengine_flex.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e # exit on any failure
+set -o pipefail # any step in pipe caused failure
+set -u # undefined variables cause exit
+
+
+SERVICE_NAME="log-java-flex-con-$(echo $ENVCTL_ID | head -c 10)"
+
+destroy() {
+  set +e
+  # delete pubsub resources
+  gcloud pubsub topics delete $SERVICE_NAME -q 2> /dev/null
+  gcloud pubsub subscriptions delete $SERVICE_NAME-subscriber -q 2> /dev/null
+  # delete container images
+  export GCR_PATH=gcr.io/$PROJECT_ID/logging:$SERVICE_NAME
+  gcloud container images delete $GCR_PATH -q --force-delete-tags 2> /dev/null
```

Comment:
I guess it's fine to keep it as is in this PR, but going forward I wonder if we can ensure these operations don't fail or if they do - we capture the output and retry.